lib/console_web/controllers/gateway_controller.ex

```diff
@@ -61,6 +61,34 @@ defmodule ConsoleWeb.GatewayController do
     end
   end
 
+  def register(conn, %{"OUI" => oui, "nonce" => nonce, "gateway" => gateway}) do
+    # receive the following from router:
+    #   nonce and OUI self explanatory
+    #   gateway["id"] is 32 bit gateway ID
+    #   gateway["public_key"] is the gateway public key router passes to console
+    #   gateway["payee_address"] is the team b58b58_address
+    #
+    # First check the following is valid:
+    #   nonce verification is valid and not messed with
+    #   OUI matches console's oui
+    #
+    # Then update gateway's public key
+    #
+    # If nonce and OUI verification fails, send back error status 404
+    # If verification succeeds, send back countersigned data
```

Comment:
How do I countersign the data? Is it part of libp2p?

Reply:
i think that part is tdb. it'll be part of the transaction system eventually, but Rahul is still working on that. so you can just put some placeholder in there for now